crates/holochain/src/conductor/space.rs

```diff
@@ -130,29 +133,62 @@ pub struct TestSpace {
 
 impl Spaces {
     /// Create a new empty set of [`DnaHash`] spaces.
-    pub fn new(config: Arc<ConductorConfig>) -> ConductorResult<Self> {
+    pub async fn new(
+        config: Arc<ConductorConfig>,
+        passphrase: sodoken::BufRead,
+    ) -> ConductorResult<Self> {
         let root_db_dir: DatabasesRootPath = config
             .data_root_path
             .clone()
             .ok_or(ConductorError::NoDataRootPath)?
             .try_into()?;
+
+        let db_key_path = root_db_dir.join("db.key");
+        let db_key = match tokio::fs::read_to_string(db_key_path.clone()).await {
+            Ok(locked) => DbKey::load(locked, passphrase).await?,
+            Err(_) => {
+                let db_key = DbKey::generate(passphrase).await?;
+                tokio::fs::write(db_key_path, db_key.locked.clone()).await?;
```

Comment:
Yes, for sure, that's something still on my list. My plan was to have an additional holochain cli flag (like `--danger-print-db-key`) which would write the list of PRAGMA commands you need to use sqlcipher to stderr. Does that sound like it would work for you?

Reply:
Yep, that sounds super

Reply:
Just to clarify. We need to write the db key file to disk. The database key is random bytes encrypted with the passphrase, so we can't generate it from the passphrase. We will need it to be included in backups and in the zipped bundles if we are to be able to decrypt databases.

Reply:
Okay, done: https://github.com/holochain/holochain/pull/4198/commits/decfe0d0cea4844b9818da1c4b9c0295a53bf2e9

without cli flag:

```
$ cargo run --manifest-path holochain/Cargo.toml --features sqlite-encrypted -- -c yo.yaml    Finished dev [unoptimized + debuginfo] target(s) in 0.40s
     Running `/home/neonphog/projects/holochain/target/debug/holochain -c yo.yaml`
Initialising log output formatting with option Log
# passphrase> 
# lair-keystore connection_url # unix:///tmp/bob/ks/socket?k=VLb4uCYPUCidb-yR84XyQs4QNKCIJMmwBHgKv381OCM #
# lair-keystore running #

###HOLOCHAIN_SETUP###
###HOLOCHAIN_SETUP_END###
Conductor ready.
```

WITH cli flag:

```
$ cargo run --manifest-path holochain/Cargo.toml --features sqlite-encrypted -- --danger-print-db-secrets -c yo.yaml
    Finished dev [unoptimized + debuginfo] target(s) in 0.39s
     Running `/home/neonphog/projects/holochain/target/debug/holochain --danger-print-db-secrets -c yo.yaml`
Initialising log output formatting with option Log
# passphrase> 
# lair-keystore connection_url # unix:///tmp/bob/ks/socket?k=VLb4uCYPUCidb-yR84XyQs4QNKCIJMmwBHgKv381OCM #
# lair-keystore running #
--beg-db-secrets--
PRAGMA key = "x'70629800023DA3CAEBBF98D2222D9C42A21C690FE5521419235C65CB4416924F'";
PRAGMA cipher_salt = "x'C9BCA2D3DED8F49BB6E0F486EC211E79'";
PRAGMA cipher_compatibility = 4;
PRAGMA cipher_plaintext_header_size = 32;
--end-db-secrets--

###HOLOCHAIN_SETUP###
###HOLOCHAIN_SETUP_END###
Conductor ready.
```